pkg/infra/httpclient/httpclientprovider/http_client_provider_test.go

```diff
@@ -49,13 +50,14 @@ func TestHTTPClientProvider(t *testing.T) {
 		_ = New(&setting.Cfg{SigV4AuthEnabled: true}, tracer)
 		require.Len(t, providerOpts, 1)
 		o := providerOpts[0]
-		require.Len(t, o.Middlewares, 7)
+		require.Len(t, o.Middlewares, 8)
 		require.Equal(t, TracingMiddlewareName, o.Middlewares[0].(sdkhttpclient.MiddlewareName).MiddlewareName())
 		require.Equal(t, DataSourceMetricsMiddlewareName, o.Middlewares[1].(sdkhttpclient.MiddlewareName).MiddlewareName())
-		require.Equal(t, SetUserAgentMiddlewareName, o.Middlewares[2].(sdkhttpclient.MiddlewareName).MiddlewareName())
-		require.Equal(t, sdkhttpclient.BasicAuthenticationMiddlewareName, o.Middlewares[3].(sdkhttpclient.MiddlewareName).MiddlewareName())
-		require.Equal(t, sdkhttpclient.CustomHeadersMiddlewareName, o.Middlewares[4].(sdkhttpclient.MiddlewareName).MiddlewareName())
-		require.Equal(t, ResponseLimitMiddlewareName, o.Middlewares[5].(sdkhttpclient.MiddlewareName).MiddlewareName())
-		require.Equal(t, SigV4MiddlewareName, o.Middlewares[6].(sdkhttpclient.MiddlewareName).MiddlewareName())
+		require.Equal(t, HTTPLoggerMiddlewareName, o.Middlewares[2].(sdkhttpclient.MiddlewareName).MiddlewareName())
+		require.Equal(t, SetUserAgentMiddlewareName, o.Middlewares[3].(sdkhttpclient.MiddlewareName).MiddlewareName())
+		require.Equal(t, sdkhttpclient.BasicAuthenticationMiddlewareName, o.Middlewares[4].(sdkhttpclient.MiddlewareName).MiddlewareName())
+		require.Equal(t, sdkhttpclient.CustomHeadersMiddlewareName, o.Middlewares[5].(sdkhttpclient.MiddlewareName).MiddlewareName())
+		require.Equal(t, ResponseLimitMiddlewareName, o.Middlewares[6].(sdkhttpclient.MiddlewareName).MiddlewareName())
+		require.Equal(t, SigV4MiddlewareName, o.Middlewares[7].(sdkhttpclient.MiddlewareName).MiddlewareName())
```

Comment:
Nit: Did you consider having a dedicated config for enabling the HTTP logger in general like SigV4? That way it won't be part of the chain by default. Another config I know 😞 , but IMO it's nice to keep the middleware chain as shallow as possible

Reply:
@wbrowne does sigv4 get applied to all requests if it's enabled?

Reply:
@toddtreece  It has to first be [enabled](https://grafana.com/docs/grafana/latest/administration/configuration/#sigv4_auth_enabled) in Grafana. Then it has to be configured in the `httpclient.Options` - this is done on a per data source basis via the `JSONData` - see [here](https://github.com/grafana/grafana/blob/main/pkg/services/datasources/service/datasource_service.go#L365-L382)

Reply:
ok, i think i've got a potential solution that avoids extra config in [7b2cea2](https://github.com/grafana/grafana/pull/46578/commits/7b2cea2554013d9aa64a85fe58518ba32d8f3937)

i still need to update this with the suggested changes from the SDK PR